src/ripple/app/ledger/Ledger.cpp

```diff
@@ -835,7 +839,8 @@ Ledger::getHashesByIndex (std::uint32_t minSeq, std::uint32_t maxSeq)
         std::pair<uint256, uint256>& hashes =
                 ret[rangeCheckedCast<std::uint32_t>(ls)];
         hashes.first.SetHexExact (lh);
-        hashes.second.SetHexExact (ph.value_or (""));
+        if (ph)
+            hashes.second.SetHexExact (*ph);
         if (!ph)
```

Comment:
1. Should now be `else` on line 844.
2. Shouldn't you be emptying `hashes.second`?  It has a value of some type...